libs/blog/roadmap/feature-roadmap/src/lib/feature-roadmap.component.ts

```
@@ -0,0 +1,214 @@
+        };
+        if (!layerMap[nodeDto.id]) {
+          layerMap[nodeDto.id] = [];
+        }
+      }
+    });
+
+    // setup clusters
+    Object.entries(clusterMap).forEach(([clusterNodeId, childrenNodeIds]) => {
+      const previousClusterNodeIdToNodeIdMap = childrenNodeIds.reduce(
+        (acc, primaryNodeId) => ({
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)